src/lib/idmap/sss_idmap.c

```diff
@@ -1228,12 +1229,17 @@ enum idmap_error_code sss_idmap_rev_offset_identity(struct sss_idmap_ctx *ctx,
 static char *normalize_casefold(const char *input, bool normalize,
                                 bool casefold)
 {
+    size_t lengthp;
+
     if (casefold) {
-        return (char *) utf8proc_NFKC_Casefold((const utf8proc_uint8_t *) input);
+        return (char *) u8_casefold((const uint8_t *)input, strlen(input) + 1, NULL,
```

Comment:
Hi,

do I understand correctly that you are using `strlen(input) + 1` to make sure that the returned casefolded output will be NUL terminated? Is this documented anywhere?

bye,
Sumit

Reply:
I'd say: to include '\0' into translation.

First I forgot '+ 1' and unit tests failed (hash input likely wasn't NULL terminated).

Imo, it's reasonable to expect that normalized and casefolded '\0' is still '\0', but I don't find it documented in https://www.gnu.org/software/libunistring/manual/libunistring.html#Case-insensitive-comparison or in https://www.gnu.org/software/libunistring/manual/libunistring.html#Conventions

Reply:
IMO, in the worst case this adds one more character to be processed but ensures that the output string contains NULL.

Reply:
Ok, see keep fingers crossed.